src/runtime.jl

```diff
@@ -154,7 +158,7 @@ end
 #     public modname
 #
 # And run statments in the toplevel expression `body`
-function eval_module(parentmod, modname, body)
+function eval_module(parentmod, modname, expr_compat_mode, body)
```

Comment:
Having `expr_compat_mode` here seems very apt, by the way.

Longer term, I imagine we store the setting in the module and use it for future `eval()`'s into that module.